adapters/Pangle/PangleAdapter/GADMediationAdapterPangle.m

```diff
@@ -13,32 +13,37 @@
 // limitations under the License.
 
 #import "GADMediationAdapterPangle.h"
-#import <BUAdSDK/BUAdSDK.h>
 #import "GADMAdapterPangleUtils.h"
 #import "GADMediationAdapterPangleConstants.h"
 #import "GADPangleRTBBannerRenderer.h"
 #import "GADPangleRTBInterstitialRenderer.h"
 #import "GADPangleRTBNativeRenderer.h"
 #import "GADPangleRTBRewardedRenderer.h"
+#import "GADPangleNetworkExtras.h"
+#import <PAGAdSDK/PAGAdSDK.h>
 
-static NSInteger _gdpr = -1, _ccpa = -1;
+static NSInteger _GDPRConsent = -1, _doNotSell = -1;
 
 @implementation GADMediationAdapterPangle {
-  /// Pangle banner ad wrapper.
-  GADPangleRTBBannerRenderer *_bannerRenderer;
-  /// Pangle interstitial ad wrapper.
-  GADPangleRTBInterstitialRenderer *_interstitialRenderer;
-  /// Pangle rewarded ad wrapper.
-  GADPangleRTBRewardedRenderer *_rewardedRenderer;
-  /// Pangle native ad wrapper.
-  GADPangleRTBNativeRenderer *_nativeRenderer;
+    /// Pangle banner ad wrapper.
+    GADPangleRTBBannerRenderer *_bannerRenderer;
+    /// Pangle interstitial ad wrapper.
+    GADPangleRTBInterstitialRenderer *_interstitialRenderer;
+    /// Pangle rewarded ad wrapper.
+    GADPangleRTBRewardedRenderer *_rewardedRenderer;
+    /// Pangle native ad wrapper.
+    GADPangleRTBNativeRenderer *_nativeRenderer;
 }
 
 - (void)collectSignalsForRequestParameters:(nonnull GADRTBRequestParameters *)params
                          completionHandler:
-                             (nonnull GADRTBSignalCompletionHandler)completionHandler {
-  NSString *signals = [BUAdSDKManager getBiddingToken:nil];
-  completionHandler(signals, nil);
+(nonnull GADRTBSignalCompletionHandler)completionHandler {
+    GADPangleNetworkExtras *extras = [params.extras isKindOfClass:[GADPangleNetworkExtras class]] ? params.extras : nil;
+    if (extras && extras.userDataString.length > 0) {
+        [PAGConfig shareConfig].userDataString = extras.userDataString;
+    }
```

Comment:
Why do we need the extras for collecting signals?

Reply:
When should the publishers set this user information? Should it be before Pangle SDK initializes or at ad request time?

Reply:
1. Custom parameters for sending Pangle the user details, like in-app purchases status. To enable this function please contact your Pangle's AM for more information 2.Publishers set this user information at ad request time.

Reply:
@wangyuanyuanyy 
This shouldn't be set here, please remove.

Like you have said, this should be set at ad request time.
Please create a utils class and a method within it, then in `render()` call that method that set the user data.

See Android PR for example:
https://github.com/googleads/googleads-mobile-android-mediation/pull/416/files#diff-b814f329c3ba459244ae4a400ca481c4ab6cfba6d8fc9341e05e90c1b0f3514eR41

Reply:
Sorry, I misunderstood what you meant. Publishers set this user information before get bidding token.

Reply:
@wangyuanyuanyy 

Follow-up(s):
- Does `[PAGSdk getBiddingToken:nil]` use that user data set in `[PAGConfig shareConfig].userDataString` to generate token (or signal)?

Reply:
Yes,`[PAGSdk getBiddingToken:nil]`user that user data set in `[PAGConfig shareConfig].userDataString` to get generate token.